File: Array/1441. Build an Array With Stack Operations/BuildArray.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<String> buildArray(int[] target, int n) {
        List<String> result = new ArrayList<>();
        int index = 1;
        for (int i = 0; i < target.length; i++) {
            while (target[i] > index) {
                result.add("Push");
                result.add("Pop");
                index++;
            }
            result.add("Push");
            index++;
        }
        return result;
    }
}

File: Array/1441. Build an Array With Stack Operations/BuildArray.go
package main

import "fmt"

func buildArray(target []int, n int) []string {
	result := []string{}
	index := 1
	for i := 0; i < len(target); i++ {
		for target[i] > index {
			result = append(result, "Push")
			result = append(result, "Pop")
			index++
		}
		result = append(result, "Push")
		index++
	}
	return result
}

func main() {
	target := []int{1, 3}
	n := 3
	result := buildArray(target, n)
	fmt.Println(result)
}
